docs/examples/notebooks/webdataset_reader_example.ipynb

```diff
@@ -0,0 +1,193 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "ccbebed3-9329-4e75-ba38-b06574891cf5",
+   "metadata": {},
+   "source": [
+    "# Webdataset reader example in rocAL\n",
+    "This example demonstrates how to set up a simple webdataset reader pipeline that loads and decodes image data stored in webdataset format using rocAL."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1227f440-4c24-4e20-9c96-9a167856b081",
+   "metadata": {},
+   "source": [
+    "<font size=\"12\"> Common Code </font>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7299da1c-e1f0-4672-8253-086b8a997b7e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from amd.rocal.plugin.generic import ROCALClassificationIterator\n",
+    "from amd.rocal.pipeline import pipeline_def\n",
+    "import amd.rocal.fn as fn\n",
+    "import amd.rocal.types as types\n",
+    "import matplotlib.pyplot as plt\n",
+    "import os\n",
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7267c8c0",
+   "metadata": {},
+   "source": [
+    "## Using fn.readers.webdataset operator\n",
+    "\n",
+    "Data stored in WebDataset format is read using `fn.readers.webdataset`. `fn.readers.webdataset` takes the following arguments:\n",
+    "\n",
+    "* `path`: the path to the webdataset tar files.\n",
+    "* `index_paths`: the path to the index files containing data about the tar files. If no path is provided, information is inferred from the tar files themselves. See `tar2idx` in the `tools` folder.\n",
+    "* `ext`: extensions of the types of data in the tar files. If there's more than one extension, the extensions must be separated by a semicolon (',').\n",
+    "* `missing_components_behavior`: specifies how the reader behaves when a sample can't return for a component for a specific output. This can be one of three options:\n",
+    "    * `MISSING_COMPONENT_EMPTY`: an empty tensor is returned.\n",
+    "    * `MISSING_COMPONENT_SKIP`: the sample is skipped.\n",
+    "    * `MISSING_COMPONENT_ERROR`: an error is thrown. This is the default behaviour.\n",
+    "`fn.readers.webdataset` also accepts arguments common to all readers: random number generator seed, shuffling, sharding, and last batch policy.\n",
+    "\n",
+    "\n",
+    "**Note**: The `ROCAL_DATA_PATH` environment variable should point to the location where data from the [MIVisionX-data repository](https://github.com/ROCm/MIVisionX-data) is downloaded."
```

Comment:
cannot link to MIVisionX-data. It's private. We did a clean up for this before. Please refer to other readmes to see how we have changed it